src/xtgeo/well/well1.py

```diff
@@ -372,11 +372,23 @@ def zonelogname(self, zname):
             self._zonelogname = None
 
     @property
+    @deprecation.deprecated(
+        deprecated_in="3.6",
+        removed_in="4.0",
+        current_version=__version__,
+        details="Use get_dataframe instead",
+    )
     def dataframe(self):
         """Returns or set the Pandas dataframe object for all logs."""
         return self._wdata.get_dataframe()
 
     @dataframe.setter
+    @deprecation.deprecated(
+        deprecated_in="3.6",
+        removed_in="4.0",
+        current_version=__version__,
+        details="Use set_dataframe instead",
+    )
     def dataframe(self, dfr):
         self.set_dataframe(dfr)  # this will include consistency checking!
```

Comment:
I think `dataframe` is used a lot for wells, so we need to discuss this. That said, using functions instead may be preferred to avoid some issues that the properties may introduce.

Reply:
I was a bit worried about that. I see that they're used in the tests quite a lot alone for these classes (+ Points, Polygons). 

My primary rationale is just that the `dataframe` property in Grid etc has had deprecation notices for `dataframe` for some time now so I thought the consistency might worth it.

Reply:
Yes the reason for that is that for Points/Polygons/Wells, a pandas dataframe is the basic "storage" in xtgeo, while for Grid/Surfaces it is numpy arrays, which I felt was unnatural to have a "dataframe" property for. 

Anyway I really like the idea of a deprecation here, but it may be that we need to delay that to xtgeo ver. 5. If we do so, we need also to change the test code and adapt Drogon. 

Some thoughts:
It would be beneficial if the `get_dataframe()` returns a copy, not a view of the underlying dataframe. Hence:

```
dfr = mywell.get_dataframe()  # return a deep copy
dfr["NEW"] = dfr["PORO"] * 2   # or some operation that creates a new log
mywell.set_dataframe(dfr)   # this will then perform all the consistency checking behind the scene

```